wdl/downsample.wdl

```diff
@@ -1,305 +1,284 @@
-    String fastq_downsample_2_name = basename(fastq_file_2, ".fastq") + "_downsample.fastq"
-
-    output {
-        File downsample_file_1 = fastq_downsample_1_name
-        File downsample_file_2 = fastq_downsample_2_name
-    }
-
-    command <<<
-
-        count1=$(bash /opt/count_fastq.sh ${fastq_file_1})
-        echo ${count1}
-        count2=$(bash /opt/count_fastq.sh ${fastq_file_2})
-        echo ${count2}
-
-        initial=$(echo $start_depth | awk ' { printf "%0.2f\n", ($1 / 2); } ')
-        final=$(echo $final_depth | awk ' { printf "%0.2f\n", ($1 / 2); } ')
-
-        if [ $count1 -eq $count2 ]
-          then
-
-            freads=$(bash /opt/calcRD.sh $initial $final $count1)  #half of the total read coverage 30x
-
-            echo "freads: " $freads
-
-            seed=$RANDOM
-            echo "seed: " $seed
-
-            fastq-sample -n ${freads} --seed ${seed} -o ${downsample_file_1} ${fastq_file_1}
-
-            fastq-sample -n ${freads} --seed ${seed} -o ${downsample_file_2} ${fastq_file_2}
-
-        else
-            echo "Error: counts don't match up!"
-            echo $count1
-            echo $count2
-        fi
-    >>>
-
-    runtime {
-        docker: downsample_docker
-    }
-
-}
-
-task realign {
+  input {
+    File reference_fasta
+    String downsample_docker
+    File bam_or_cram_file
+
+    File ref_amb
+    File ref_ann
+    File ref_bwt
+    File ref_pac
+    File ref_sa
+    File ref_fai
+    File ref_dict
+
+    String sample_ID
+    File intervals_exons
+
+    #default start depth is 30x
+    Float start_depth = 30
+    Int? seed_override  
+
+    File? gatk4_jar_override
+    String gatk_docker
+
+    # Runtime configuration overrides
+    RuntimeAttr? runtime_attr_cram_to_bam
+    RuntimeAttr? runtime_attr_bam_to_fastq 
+    RuntimeAttr? runtime_attr_random_sample
+    RuntimeAttr? runtime_attr_realign
+    RuntimeAttr? runtime_attr_add_read_group
+    RuntimeAttr? runtime_attr_mark_duplicates
+    RuntimeAttr? runtime_attr_sort_index
+    RuntimeAttr? runtime_attr_count_coverage
+    RuntimeAttr? runtime_attr_collect_counts   
+
+    #Execution defaults and overrides
+    Boolean run_downsample_2x = true
+    Boolean run_downsample_4x = true
+    Boolean run_downsample_6x = true
+    Boolean run_downsample_8x = true
+    Boolean run_downsample_custom = false
+
+    Float? final_depth_custom
+
+  }
+
+  meta {
+    author: "Stephanie Hao"
+    email: "shao@broadinstitute.org"
+  }
     
-    input {
-        File downsample_file_1 
-        File downsample_file_2 
-        File reference_fasta
-        String downsample_docker
+  #################################################################################
+  ####        Calls part 1 to convert cram/bam file to paired fastq               #
+  #################################################################################
+
+  call ds1.downSampling_01 as downSampling_01 {
```

Comment:
Copied (& edited to match this line) from a comment I got from Mark: 

I know we do this a lot in the code but we actually shouldn't be assigning an alias explicitly. It turns out that even if you are calling from an imported WDL, the `as downSampling_01` is unnecessary. Outputs will still have the syntax `downSampling_01.out_var` if you use just call `ds1.downSampling_01`. Only need to use `as` if you have multiple calls to the same task (like with ds2).